### ISSUE TITLE
Some Issues Regarding Code Replication

### ISSUE THREAD

**Issue by seamoke** [Kipok/NeMo-Skills#21]
Thank you for sharing the dataset. 
I retrained the OpenMath dataset using Mistral-7B, but the accuracy for GSM8K is only 64% and for Math it is 24%. However, when using the nvidia/OpenMath-Mistral-7B-v0.1-hf from Hugging Face, the accuracy indeed reaches 80%. 
I use the same praramter: 
epoch=4
learning rate=1e-6
The training loss is as follow:
![W B Chart 2024_4_24 11_55_32](https://github.com/Kipok/NeMo-Skills/assets/51767292/e2d0f3d9-24a2-4288-bdae-3f93cf7b8b93)
![W B Chart 2024_4_24 11_55_44](https://github.com/Kipok/NeMo-Skills/assets/51767292/b2b355de-3a2a-46e8-9a78-328745e1c586)

After 5000 steps, the performance remains almost unchanged, with GSM8K at 64 and Math at 24. ( I have tested checkpoint_5000, 10000, 15000)
Could you help me?

**Comment by Kipok** [comment]
@seamoke can you please add a couple of details to make it easier for us to help you. 
- Did you use the latest commit or v0.1 tag (the sft container was upgraded to the newer nemo version and it might have some changes to the training results, although it shouldn't be this drastic)? 
- To double check - you're using our provided config https://github.com/Kipok/NeMo-Skills/blob/main/nemo_skills/finetuning/sft_config.yaml and only replace epoch/lr, but other parameters are unchanged? 
- Can you also share the plot of the validation accuracy (should be enabled by default, unless you changed that)?

**Comment by Kipok** [comment]
@seamoke are you still having issues? Going to close the issue if we don't get a reply from you

**Comment by seamoke** [comment]
> @seamoke are you still having issues? Going to close the issue if we don't get a reply from you

Thanks for your reply. Because I use huggingface's Trainer to finetune model. So it's hars to compare the difference. I will close the issue and check my code again.

**Comment by Kipok** [comment]
Sounds good! By the way, if you can share - what's the reason why you prefer huggingface's trainer to the provided finetuning workflow that's using nemo?

**Comment by seamoke** [comment]
> Sounds good! By the way, if you can share - what's the reason why you prefer huggingface's trainer to the provided finetuning workflow that's using nemo?

just I have some other code in that repo. So I use Trainer to finetune.
